torch/_dynamo/variables/higher_order_ops.py

```diff
@@ -54,6 +54,49 @@ def are_tensors(var):
     return False
 
 
+def check_args_and_maybe_add_to_graph(sub_args, tracer, tx, add_arg_to_graph):
```

Comment:
Help with naming

Reply:
We probably want to split this function into `validate_args` and `manually_create_graph_inputs` ? As written the function looks like it is doing too much

Reply:
I initially did that, but I didn't like it because each type requires it's own handling. So later if we want to support a new type, let's say `SymNodeVariable` then it would have to be updated for both functions, which seemed bad. Wdyt?

Reply:
Up to you, in the end state we'll delete `manually_create_graph_inputs` after moving all the other HigherOrderOperator to `add_arg_to_subgraph=False`. If we keep it as is then my suggested name would be something like `validate_args_and_manually_create_graph_inputs`

Reply:
Noted! Will update the name for now and clean-up this function. Thanks!